ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/config/NTupleOmitPrefixStorageLayoutConfig.java

```diff
@@ -0,0 +1,180 @@
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.commons.lang3.StringUtils;
+
+import edu.wisc.library.ocfl.core.extension.storage.layout.NTupleOmitPrefixStorageLayoutExtension;
+import edu.wisc.library.ocfl.core.extension.OcflExtensionConfig;
+
+/**
+ * Configuration for the <a href="https://ocfl.github.io/extensions/0007-n-tuple-omit-prefix-storage-layout.md">
+ * N Tuple Storage Layout</a> extension.
+ *
+ * @author vcrema
+ * @since 2021-10-25
+ */
+public class NTupleOmitPrefixStorageLayoutConfig implements OcflExtensionConfig {
+
+	public static final String ZERO_PADDING_LEFT = "left";
+	public static final String ZERO_PADDING_RIGHT = "right";
+	public static final String ZERO_PADDING_NONE = "none";
+
+	private String delimiter;
+	private int tupleSize = 3;
+	private int numberOfTuples = 3;
+	private String zeroPadding = ZERO_PADDING_LEFT;
+	private boolean reverseObjectRoot = false;
+
+	@JsonIgnore
+	@Override
+	public String getExtensionName() {
+		return NTupleOmitPrefixStorageLayoutExtension.EXTENSION_NAME;
+	}
+
+	@JsonIgnore
+	@Override
+	public boolean hasParameters() {
+		return true;
+	}
+
+	/**
+	 * @return the delimiter marking end of prefix
+	 */
+	public String getDelimiter() {
+		return delimiter;
+	}
+
+	/**
+	 * @return tupleSize - the segment size (in characters) to split the digest
+	 *         into
+	 */
+	public int getTupleSize() {
+		return tupleSize;
+	}
+
+	/**
+	 * @return The number of segments to use for path generation
+	 */
+	public int getNumberOfTuples() {
+		return numberOfTuples;
+	}
+
+	/**
+	 * return zeroPadding - Indicates whether to use left or right zero padding
+	 * for ids less than tupleSize * numberOfTuples
+	 */
+	public String getZeroPadding() {
+		return zeroPadding;
+	}
+
+	/**
+	 * return true or false, indicates that the prefix-omitted, padded object
+	 * identifier should be reversed
+	 */
+	public boolean reverseObjectRoot() {
+		return reverseObjectRoot;
+	}
+
+	/**
+	 * The case-insensitive, delimiter marking the end of the OCFL object
+	 * identifier prefix; MUST consist of a character string of length one or
+	 * greater. If the delimiter is found multiple times in the OCFL object
+	 * identifier, its last occurrence (right-most) will be used to select the
+	 * termination of the prefix.
+	 *
+	 * @param delimiter
+	 *            marking the end of prefix
+	 */
+	public NTupleOmitPrefixStorageLayoutConfig setDelimiter(String delimiter) {
+		if (StringUtils.isBlank(delimiter)) {
+			throw new IllegalArgumentException("Arg must not be empty or null: 'delimiter'");
+		}
+		this.delimiter = delimiter;
+		return this;
+	}
+
+	/**
+	 * the segment size (in characters) to split the digest into
+	 *
+	 * @param tupleSize
+	 *            - the segment size (in characters) to split the digest into
+	 * 
+	 */
+	public NTupleOmitPrefixStorageLayoutConfig setTupleSize(int tupleSize) {
+		if (tupleSize <= 0) {
```

Comment:
The extension defines this as being between 1 and 32 inclusive. So, this could be rewritten as:

```java
this.tupleSize = Enforce.expressionTrue(tupleSize >=1 && tupleSize <= 32,
                tupleSize, "tupleSize must be between 1 and 32 inclusive");
```